soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -17,6 +20,648 @@
 #include "soh/Enhancements/game-interactor/GameInteractor_Hooks.h"
 #include <assert.h>
 
+typedef struct {
+    s16 left;
+    s16 top;
+} IconPosition;
+
+typedef struct {
+    s16 width;
+    s16 height;
+} IconSize;
+
+typedef struct {
+    Sprite sprite;
+    Color_RGBA8 color;
+    u8 item;
+    IconPosition pos;
+    IconSize size;
+} ItemData;
+
+#define CREATE_SPRITE_24(iconTex, spriteId) { iconTex, 24, 24, G_IM_FMT_RGBA, G_IM_SIZ_32b, spriteId }, {0xFF, 0xFF, 0xFF, 0xFF}
+#define CREATE_SPRITE_32(iconTex, spriteId) { iconTex, 32, 32, G_IM_FMT_RGBA, G_IM_SIZ_32b, spriteId }, {0xFF, 0xFF, 0xFF, 0xFF}
+#define CREATE_SPRITE_SONG(colorR, colorG, colorB) { dgSongNoteTex, 16, 24, G_IM_FMT_IA, G_IM_SIZ_8b, 100 }, {colorR, colorG, colorB, 0xFF}
+#define CREATE_SPRITE_RUPEE(colorR, colorG, colorB) { dgRupeeCounterIconTex, 16, 16, G_IM_FMT_IA, G_IM_SIZ_8b, 102 }, {colorR, colorG, colorB, 0xFF}
+#define CREATE_SPRITE_SKULL { dgDungeonMapSkullTex, 16, 16, G_IM_FMT_RGBA, G_IM_SIZ_16b, 104 }, {0xFF, 0xFF, 0xFF, 0xFF}
+#define CREATE_SPRITE_COUNTER_DIGIT(i) { dgAmmoDigit##i##Tex, 8, 8, G_IM_FMT_IA, G_IM_SIZ_8b, 105+i }
+
+#define ICON_SIZE 12
+#define COUNTER_SIZE 16
+#define SONG_WIDTH 8
+#define SONG_HEIGHT 12
+
+#define LEFT_OFFSET (int)0x37
+#define TOP_OFFSET  (int)0x5C
+
+#define COUNTER_DIGITS_LEFT_OFFSET COUNTER_SIZE / 2 - 3
+#define COUNTER_DIGITS_TOP_OFFSET COUNTER_SIZE - 3
+
+#define SIZE_NORMAL {ICON_SIZE, ICON_SIZE}
+#define SIZE_COUNTER {COUNTER_SIZE, COUNTER_SIZE}
+#define SIZE_SONG {SONG_WIDTH, SONG_HEIGHT}
+
+#define INV_IC_POS(x, y) {0x4E + ICON_SIZE * x, 0x00 + ICON_SIZE * y}
+#define EQP_IC_POS(x, y) {0x7E + ICON_SIZE * x, 0x2A + ICON_SIZE * y}
+#define SNG_IC_POS(x, y) {0x49 + SONG_WIDTH * x, 0x45 + SONG_HEIGHT * y}
+#define UPG_IC_POS(x, y) {0x5A + ICON_SIZE * x, 0x2A + ICON_SIZE * y}
+#define STN_IC_POS(i) {0x29 + ICON_SIZE * i, 0x31}
```

Comment:
nit: it'd be nice to have these named without abbreviations

https://www.youtube.com/watch?v=-J3wNP6u5YU&t=68s